### PR TITLE
feat(api): allow multiple API/Gateway addresses

### PR DESCRIPTION
**Pull request by overbool**
Refer to: https://github.com/ipfs/go-ipfs-config/pull/11
Fixes: https://github.com/ipfs/go-ipfs/issues/5401

License: MIT
Signed-off-by: Overbool <overbool.xu@gmail.com>

### CODE REVIEW

**Comment by overbool** [comment]
@Stebalien could u help me review it? refer to https://github.com/ipfs/go-ipfs-config/pull/11

File: cmd/ipfs/daemon.go [hunk 1]
```diff
@@ -461,15 +471,25 @@ func serveHTTPApi(req *cmds.Request, cctx *oldcmds.Context) (<-chan error, error
 		return nil, fmt.Errorf("serveHTTPApi: ConstructNode() failed: %s", err)
 	}
 
-	if err := node.Repo.SetAPIAddr(apiMaddr); err != nil {
+	if err := node.Repo.SetAPIAddr(listeners[0].Multiaddr()); err != nil {
```
Comment (hannahhoward) [inline cmd/ipfs/daemon.go]: do we want to connect the repo to only one of the addresses? Should it always be the first?

Comment (Stebalien) [reply cmd/ipfs/daemon.go]: So, adding support for multiple addrs here would require changing the `~/.ipfs/api` file format (or, more likely, introducing a *second* API file with support for multiple addresses). Choosing the *first* address is probably reasonable for now. Eventually, we may want to consider preferring a loopback address, or something like that, but we can revisit this later.

**Comment by Stebalien** [comment]
Blocked on go-ipfs-config update (in progress).

**Comment by lgierth** [comment]
This has been merged as part of #5553 -- thank you :]

**Pull request closed without merging**